lib/kafka/sasl/awsmskiam.rb

```diff
@@ -68,11 +113,11 @@ def authentication_payload(host:, time_now:)
           'user-agent' => "ruby-kafka",
           'action' => "kafka-cluster:Connect",
           'x-amz-algorithm' => "AWS4-HMAC-SHA256",
-          'x-amz-credential' => @access_key_id + "/" + time_now.strftime("%Y%m%d") + "/" + @aws_region + "/kafka-cluster/aws4_request",
+          'x-amz-credential' => @aws_msk_iam_credentials.get_access_key_id + "/" + time_now.strftime("%Y%m%d") + "/" + @aws_region + "/kafka-cluster/aws4_request",
           'x-amz-date' => time_now.strftime("%Y%m%dT%H%M%SZ"),
           'x-amz-signedheaders' => "host",
           'x-amz-expires' => "900",
-          'x-amz-security-token' => @session_token,
+          'x-amz-security-token' => @aws_msk_iam_credentials.get_session_token,
```

Comment:
see below. it should be acceptable to not pass a session token if you are using permanent credentials. in that case, we don't want to set this parameter (i think it'll fail if we set it to `nil` but want to confirm that that's the behavior).

Reply:
Yes, I think you are right.

When I test the existing implementation, I did notice that I have to pass `session_token` when I use static credentials. I didn't realize it was caused by setting 'x-amz-security-token' as `nil`... Let me try to set `x-amz-security-token` as optional

Reply:
I think that is a good point. `.delete_if { |_, v| v.nil? }` should be able to remove any keypair if the value is nil